cobigen-plugins/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/main/java/com/devonfw/cobigen/openapiplugin/matcher/OpenAPIMatcher.java

```diff
@@ -141,12 +143,15 @@ public boolean matches(MatcherTo matcher) {
    * @param key to search in the Map
    * @return the value of that key, and if nothing was found, an empty string
    */
-  private String getExtendedProperty(Map<String, Object> extensionProperties, String key) {
+  private String getExtendedProperty(Map<String, Object> extensionProperties, String key, GenerationReportTo report) {
 
     Map<String, Object> properties = extensionProperties;
     if (properties.containsKey(key)) {
       return properties.get(key).toString();
     } else {
+      String warningMessage = "The property " + key
+          + " was requested in a variable assignment although the input does not provide this property. Setting it to empty";
+      report.addWarning(warningMessage);
```

Comment:
Logging the warning is good, but not sure, how this will be shown to the user. Cannot see any code.
Also, it's not handled that this leads to a hard stop on the generation. Possibly you need to mark the variableAssignment to be mandatory or something like that and if so report an error instead of an warning and stop generation reporting the actual failure.